src/components/Counter/Counter.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import classNames from 'classnames';
+
+const buttonsName = ['add 1', 'add 100', 'mix and devide 5'];
+
+export class Counter extends React.Component {
+    state = {
+      counter: 0,
+      buttons: buttonsName,
+    };
+  
+    checkButton = (buttonName) => {
+      if (buttonName === this.state.buttons[0]) {
+        this.addOne();
+      } else if (buttonName === this.state.buttons[1]) {
+        this.add100();
+      } else if (buttonName === this.state.buttons[2]) {
+        this.increase();
+      }
+    }
+
+    addOne = () => {
+      this.setState({counter: this.state.counter += 1});
```

Comment:
```suggestion
      this.setState(state => ({
        counter: state.counter + 1,
      }));
```
аналогично в следующих функциях: Use callback in setState when referencing the previous state.